api/Updated-Metrics.md

```diff
@@ -0,0 +1,368 @@
+For example, the user might not be interested in the difference between the maximum and minimum, but the difference from the overall value.
+Or perhaps the difference from a particular group.
+
+### Proposed Change
+
+The `GroupedMetric` object would have methods for calculating the required scalars.
+First, let us consider the differences.
+
+We would provide operations to calculate differences in various ways (all of these results are a Pandas Series):
+```python
+>>> result.differences()
+Male      0.0
+Female    0.4406
+Name: TBD dtype: float64
+>>> result.differences(relative_to='min')
+Male     -0.4406
+Female    0.0
+Name: TBD dtype: float64
+>>> result.differences(relative_to='min', abs=True)
+Male      0.4406
+Female    0.0
+Name: TBD dtype: float64
+>>> result.differences(relative_to='overall')
+Male     -0.2436
+Female    0.1870
+Name: TBD dtype: float64
+>>> result.differences(relative_to='overall', abs=True)
+Male      0.2436
+Female    0.1870
+Name: TBD dtype: float64
+>>> result.differences(relative_to='group', group='Female', abs=True)
+Male      0.4406
+Female    0.0
+Name: TBD dtype: float64
+```
+The arguments introduced so far for the `differences()` method:
+- `relative_to=` to decide the common point for the differences. Possible values are `'max'` (the default), `'min'`, `'overall'` and `'group'`
+- `group=` to select a group name, only when `relative_to` is set to `'group'`. Default is `None`
+- `abs` to indicate whether to take the absolute value of each entry (defaults to false)
+
+The user could then use the Pandas methods `max()` and `min()` to reduce these Series objects to scalars.
+However, this will run into issues where the `relative_to` argument ends up pointing to either the maximum or minimum group, which will have a difference of zero.
+That could then be the maximum or minimum value of the set of difference, but probably won't be what the user wants.
+
+To address this case, we should add an extra argument `aggregate=` to the `differences()` method:
+```python
+>>> result.differences(aggregate='max')
+0.4406
+>>> result.differences(relative_to='overall', aggregate='max')
+0.1870
+>>> result.differences(relative_to='overall', abs=True, aggregate='max')
+0.2436
+```
+If `aggregate=None` (which would be the default), then the result is a Series, as shown above.
+
+There would be a similar method called `ratios()` on the `GroupedMetric` object:
+```python
+>>> result.ratios()
+Male      1.0
+Female    0.3259
+Name: TBD dtype: float64
+```
+The `ratios()` method will take the following arguments:
+- `relative_to=` similar to `differences()`
+- `group=` similar to `differences()`
+- `ratio_order=` determines how to build the ratio. Values are
+   - `sub_unity` to make larger value the denominator
+   - `super_unity` to make larger value the numerator
+   - `from_relative` to make the value specified by `relative_to=` the denominator
```

Comment:
both this line and the next say `relative_to`, is that intended?

Reply:
Yes. The change is between numerator and denominator.

Reply:
Is this a question of naming, or whether we need all the cases? I'm not overly happy with the names myself, but I do think that all four cases might be useful.

Reply:
A bit of both, I guess... but I think you're right that you need all four cases.